lib/echo/echo_client.rb

```diff
@@ -34,10 +34,39 @@ def get_phones(user_id, token)
       get("/echo-rest/users/#{user_id}/phones.json", {}, token_header(token))
     end
 
-    def get_preferences(user_id, token)
+    def general_contact_valid?(contact)
+      return false if contact.blank? || contact['phones'].blank? || contact['address'].blank?
+      true
+    end
+
+    def get_preferences(user_id, token, client, access_token)
       response = get("/echo-rest/users/#{user_id}/preferences.json", {}, token_header(token))
       if response.status == 404
         response = update_preferences(user_id, {preferences: {}}, token)
+      elsif response.status == 200 && !general_contact_valid?(response.body['preferences']['general_contact'])
```

Comment:
There's a slight problem here. On the configuration page, we're showing contact info coming from URS even if the user has ECHO contact info. If we then use the ECHO contact info (which could be different than the URS contact info) the user may get emails sent to an unexpected address or something of that sort. I'd say always use the URS contact info and override any ECHO contact info.